modules/services/window-managers/hyprland.nix

```
@@ -0,0 +1,120 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.wayland.windowManager.hyprland;
+  defaultHyprlandPackage = pkgs.hyprland.override {
+    enableXWayland = cfg.xwayland.enable;
+    hidpiXWayland = cfg.xwayland.hidpi;
+    nvidiaPatches = cfg.nvidiaPatches;
+  };
+in {
+  meta.maintainers = [ lib.maintainers.fufexan ];
+
+  options.wayland.windowManager.hyprland = {
+    enable = lib.mkEnableOption (lib.mdDoc "Hyprland wayland compositor");
+
+    package = lib.mkOption {
+      type = with lib.types; nullOr package;
+      default = defaultHyprlandPackage;
+      description = lib.mdDoc ''
+        Hyprland package to use. Will override the 'xwayland' and 'nvidiaPatches'
+        options.
+
+        Defaults to the one provided by the flake. Set it to
+        {package}`pkgs.hyprland` to use the one provided by nixpkgs or
+        if you have an overlay.
+
+        Set to null to not add any Hyprland package to your path. This should
+        be done if you want to use the NixOS module to install Hyprland.
+      '';
+    };
+
+    systemdIntegration = lib.mkOption {
+      type = lib.types.bool;
+      default = pkgs.stdenv.isLinux;
+      description = lib.mdDoc ''
+        Whether to enable {path}`hyprland-session.target` on
+        hyprland startup. This links to `graphical-session.target`.
+        Some important environment variables will be imported to systemd
+        and dbus user environment before reaching the target, including
+        - `DISPLAY`
+        - `HYPRLAND_INSTANCE_SIGNATURE`
+        - `WAYLAND_DISPLAY`
+        - `XDG_CURRENT_DESKTOP`
+      '';
+    };
+
+    disableAutoreload = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = lib.literalExpression "true";
+      description = lib.mdDoc ''
+        Whether to disable automatically reloading Hyprland's configuration when
+        rebuilding the Home Manager profile.
+      '';
+    };
+
+    xwayland = {
+      enable = lib.mkEnableOption (lib.mdDoc "XWayland") // { default = true; };
+      hidpi = lib.mkEnableOption null // {
+        description = lib.mdDoc ''
+          Enable HiDPI XWayland, based on [XWayland MR 733](https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/733).
+          See <https://wiki.hyprland.org/Nix/Options-Overrides/#xwayland-hidpi> for more info.
+        '';
+      };
+    };
+
+    nvidiaPatches = lib.mkEnableOption
+      (lib.mdDoc "patching wlroots for better Nvidia support.");
+
+    extraConfig = lib.mkOption {
+      type = lib.types.nullOr lib.types.lines;
+      default = "";
+      description = lib.mdDoc ''
+        Extra configuration lines to add to `~/.config/hypr/hyprland.conf`.
+      '';
+    };
+
+    recommendedEnvironment = lib.mkEnableOption null // {
+      description = lib.mdDoc ''
+        Whether to set the recommended environment variables.
+      '';
```

There are a lot of "recommended" variables for wayland (e.g. [here](https://github.com/FlorianFranzen/nixrc/blob/d3d57da19d29e9f91bd2076127b2a56a13f61a5f/pkgs/sway.nix#L15-L62) is a list of mine) so it would help to add some more info to the description.

```suggestion
      description = lib.mdDoc ''
        Whether to set the recommended environment variables (i.e. `NIXOS_OZONE_WL`).
      '';
```

---

If these aren't specific to Hyprland, maybe it would make sense to expose that option as `wayland.recommendedEnvironment` instead (in a separate module).

---

This option does not exist in the NixOS module (discussion [here](https://github.com/NixOS/nixpkgs/pull/221730#discussion_r1157860002)), so I will also remove it from here and follow up with a PR as per @ncfavier's suggestion.

---

Also note the following paragraph in the NixOS 22.05 release notes:

> If you are using Wayland you can choose to use the Ozone Wayland support
  in Chrome and several Electron apps by setting the environment variable
  `NIXOS_OZONE_WL=1` (for example via
  `environment.sessionVariables.NIXOS_OZONE_WL = "1"`).
  This is not enabled by default because Ozone Wayland is
  still under heavy development and behavior is not always flawless.
  Furthermore, not all Electron apps use the latest Electron versions.

I don't know if this is still relevant.

---

Resolved, thanks.